tfhe/src/integer/server_key/radix_parallel/tests.rs

```diff
@@ -866,98 +858,71 @@ fn integer_default_scalar_left_shift(param: Parameters) {
     //RNG
     let mut rng = rand::thread_rng();
 
-    // message_modulus^vec_length
     let modulus = param.message_modulus.0.pow(NB_CTXT as u32) as u64;
 
-    //Nb of bits to shift
-    let tmp_f64 = param.message_modulus.0 as f64;
-    let nb_bits = tmp_f64.log2().floor() as usize * NB_CTXT;
-
     for _ in 0..NB_TEST {
         let clear = rng.gen::<u64>() % modulus;
+        let scalar = rng.gen::<u32>();
 
-        let scalar = rng.gen::<usize>() % nb_bits;
-
-        // encryption of an integer
         let ct = cks.encrypt(clear);
 
-        let ct_res = sks.scalar_left_shift_parallelized(&ct, scalar);
-        let tmp = sks.scalar_left_shift_parallelized(&ct, scalar);
+        let ct_res = sks.scalar_left_shift_parallelized(&ct, scalar as u64);
+        let tmp = sks.scalar_left_shift_parallelized(&ct, scalar as u64);
         assert!(ct_res.block_carries_are_empty());
         assert_eq!(ct_res, tmp);
 
-        // decryption of ct_res
         let dec_res: u64 = cks.decrypt(&ct_res);
 
-        // assert
-        assert_eq!((clear << scalar) % modulus, dec_res);
+        assert_eq!(clear.checked_shl(scalar).unwrap_or(0) % modulus, dec_res);
     }
 }
 
 fn integer_unchecked_scalar_right_shift(param: Parameters) {
     let (cks, sks) = KEY_CACHE.get_from_params(param);
     let cks = RadixClientKey::from((cks, NB_CTXT));
 
-    //RNG
     let mut rng = rand::thread_rng();
 
     // message_modulus^vec_length
     let modulus = param.message_modulus.0.pow(NB_CTXT as u32) as u64;
 
-    //Nb of bits to shift
-    let tmp_f64 = param.message_modulus.0 as f64;
-    let nb_bits = tmp_f64.log2().floor() as usize * NB_CTXT;
-
     for _ in 0..NB_TEST {
         let clear = rng.gen::<u64>() % modulus;
+        let scalar = rng.gen::<u32>();
 
-        let scalar = rng.gen::<usize>() % nb_bits;
-
-        // encryption of an integer
         let ct = cks.encrypt(clear);
 
-        let ct_res = sks.unchecked_scalar_right_shift_parallelized(&ct, scalar);
+        let ct_res = sks.unchecked_scalar_right_shift_parallelized(&ct, scalar as u64);
 
-        // decryption of ct_res
         let dec_res: u64 = cks.decrypt(&ct_res);
 
-        // assert
-        assert_eq!(clear >> scalar, dec_res);
+        assert_eq!(clear.checked_shr(scalar).unwrap_or(0) % modulus, dec_res);
     }
 }
 
 fn integer_default_scalar_right_shift(param: Parameters) {
     let (cks, sks) = KEY_CACHE.get_from_params(param);
     let cks = RadixClientKey::from((cks, NB_CTXT));
 
-    //RNG
     let mut rng = rand::thread_rng();
 
     // message_modulus^vec_length
     let modulus = param.message_modulus.0.pow(NB_CTXT as u32) as u64;
 
-    //Nb of bits to shift
-    let tmp_f64 = param.message_modulus.0 as f64;
-    let nb_bits = tmp_f64.log2().floor() as usize * NB_CTXT;
-
     for _ in 0..NB_TEST {
         let clear = rng.gen::<u64>() % modulus;
+        let scalar = rng.gen::<u32>();
```

Comment:
sorry to be annoying but here most of the time we would be testing the oveflowing scalar shift, can we have two halves to this test ? one where we know we are not overflowing/underflowing and the other where we force the overflow/underflow ?

potentially being sure we test "in ciphertext shifts" by taking shift in 0..message_modulus would be good here

Reply:
I think we have seen in the past that corner cases are an important part of our testing and here I don't think we are covering the different shift behaviors

Reply:
should this be in to different test functions or in the same one ?

Reply:
it could be the same test function as long as we have the 3/4 sub cases well defined

Reply:
could be in their own blocks like
//Generic case
{

}

// Overlow case
{

}

// All small shifts
{
for shift in 0..message_modulus or carry_modulus {
}

Reply:
testing for all small shift for each test will increase time required to test, is that ok ?

Reply:
hmm maybe we won't loop for those (like no need to repeat the test 10 times) ? I let you be the judge of that

Reply:
I put the small shift outside the main test loop